libs/calib/calib.c

```diff
@@ -250,168 +250,241 @@ static int calib_magironRead(FILE *file, calib_data_t *cal)
 }
 
 
-static void calib_motlinDefaults(calib_data_t *cal)
+/* Returns 0 if new value `val` was successfully saved to `cal` as `paramName` parameter. -1 otherwise */
+static int calib_accorthEnter(const char *paramName, calib_data_t *cal, float val)
 {
-	cal->params.motlin.motorEq[0][0] = 0.968600;
-	cal->params.motlin.motorEq[0][1] = 0.034796;
+	matrix_t *mat = NULL;
+	unsigned int row, col;
+	float *slot;
 
-	cal->params.motlin.motorEq[1][0] = 1.031400;
-	cal->params.motlin.motorEq[1][1] = 0.085204;
+	if (strlen(paramName) != 3) {
+		return -1;
+	}
 
-	cal->params.motlin.motorEq[2][0] = 1.003427;
-	cal->params.motlin.motorEq[2][1] = 0.142546;
+	row = (uint8_t)(paramName[1] - '0'); /* convert character to unsigned int digit */
+	col = (uint8_t)(paramName[2] - '0'); /* convert character to unsigned int digit */
 
-	cal->params.motlin.motorEq[3][0] = 0.996573;
-	cal->params.motlin.motorEq[3][1] = 0.137454;
-}
+	/* matrix type get through character check */
+	switch (paramName[0]) {
+		case ACC_CHAR_ORTHO:
+			if (row > (unsigned int)'9' || col > (unsigned int)'9') {
```

Comment:
You compare to 0x39. It is ok?
Same below.
```suggestion
			if (row > 9 || col > 9) {
```

Reply:
Oh... yeah that's bad. This exact construction is used elsewhere throughout the file. I will correct all instances. Thanks for catching it!